### PR TITLE
adds onViewabilityItemChanged to inifinite scroll list

### PR DESCRIPTION
**Pull request by intergalacticspacehighway**
# Why
- We need to add viewability tracking to list to support autoplay video, mute video improvements and backend viewed NFT API integration. 
- FlashList already supports this. This PR adds it to the Virtuoso list on web. API is kept same as FlashList/FlatList

# How
- Use IntersectionObserver as Virtuoso doesn't natively support this as mentioned here https://github.com/petyosi/react-virtuoso/issues/118#issuecomment-642156138. 
- Currently, kept it simple i.e. 70% visible item is considered as visible. 

# Test Plan
- Passing `onViewabilityItemsChanged` should trigger when items are visible

### CODE REVIEW

File: packages/app/components/claim/claim-form.tsx [hunk 1]
```diff
@@ -61,8 +61,12 @@ export const ClaimForm = ({ edition }: { edition: CreatorEditionResponse }) => {
     nft?.data.item.creator_airdrop_edition_address
   );
 
+  const { user } = useUser();
   const handleClaimNFT = async () => {
-    if (nft?.data.item.creator_id) {
+    if (
+      nft?.data.item.creator_id &&
+      user?.data?.profile.profile_id !== nft?.data.item.creator_id
```
Comment (intergalacticspacehighway) [inline packages/app/components/claim/claim-form.tsx:68]: fixed a bug that led to follow your own profile when claiming your own drop

File: packages/app/lib/infinite-scroll-list/index.web.tsx [hunk 2]
```diff
@@ -1,31 +1,94 @@
...
   onEndReached?: VirtuosoGridProps["endReached"];
+  onViewableItemsChanged?: FlatListProps<any>["onViewableItemsChanged"];
 };
 const renderComponent = (Component: any) => {
   if (!Component) return null;
   if (React.isValidElement(Component)) return Component;
   return <Component />;
 };
 
+const ViewablityTracker = ({
```
Comment (axeldelafosse) [inline packages/app/lib/infinite-scroll-list/index.web.tsx]: `ViewabilityTracker`

**Review by axeldelafosse — APPROVED** [review]
LFG ✨

**Review by alantoa — APPROVED** [review]
LGTM! ✨

**Pull request merged**